### ISSUE TITLE
Use `pyproject.toml` instead of `setup.cfg` and `setup.py`

### ISSUE THREAD

**Issue by mbalatsko** [raimon49/pip-licenses#201]
I suggest upgrading the project structure to a more modern python package format using `pyproject.toml`. I could gladly get it done, if you agree @raimon49. But I wanted to clarify some things first:

- Is it a hard requirement for the package to have `pip-compile` as the packaging manager? Poetry seems to be able to handle all transitive dependencies properly and does not require exact versions specified, it works well with version bounds. For downstream packages, `==` in requirements could become substantial issue
- Are you ok with having version, author, etc. only in pyproject.toml?

**Comment by stefan6419846** [comment]
`pip-tools`, where `pip-compile` is part of, is just used for having a known set of fixed versions. The packaging backend already is `setuptools` and the versions are not really fixed inside the configuration file (although I am not sure why `pytest-runner` would be required for building the package itself): https://github.com/raimon49/pip-licenses/blob/b3a3d8739e2b725e39d142283d66d0c5f320b3d4/setup.cfg#L27-L37

**Comment by raimon49** [comment]
Thanks for the suggestion, I also worked on another project to migrate to `pyproject.toml` but stopped halfway. At that time, I selected `setuptools>=61.0`, build_meta as `build-backend`.

The main toolchain used in this project is as commented by @stefan6419846 . I know that tools like Pipenv and Poetry have appeared, but I am still using pip-tools+venv. The Makefile for this project describes the main tasks for the release.

In moving to `pyproject.toml`, I have no problem if metadata such as version, author, classifiers are written only in `pyproject.toml`.
